File: admin/src/api/mall_trade_delivery_express_template.ts
import { PaginatedRequest, PaginatedResponse } from "@/base/page";
import { api } from "@/utils/request";
import { MallTradeDeliveryExpressTemplateChargeRequest, MallTradeDeliveryExpressTemplateChargeResponse } from "./mall_trade_delivery_express_template_charge";
import { MallTradeDeliveryExpressTemplateFreeRequest, MallTradeDeliveryExpressTemplateFreeResponse } from "./mall_trade_delivery_express_template_free";

const apis = {
  create: "/mall/mall_trade_delivery_express_template/create", // 新增
  update: "/mall/mall_trade_delivery_express_template/update", // 修改
  delete: "/mall/mall_trade_delivery_express_template/delete", // 删除
  get: "/mall/mall_trade_delivery_express_template/get", // 单条查询
  get_base: "/mall/mall_trade_delivery_express_template/get_base", // 单条查询
  list: "/mall/mall_trade_delivery_express_template/list", // 列表查询
  page: "/mall/mall_trade_delivery_express_template/page", // 分页查询
  enable: "/mall/mall_trade_delivery_express_template/enable", // 启用
  disable: "/mall/mall_trade_delivery_express_template/disable", // 禁用
};

export interface MallTradeDeliveryExpressTemplateRequest {
  id: number; // 编号
  name: string; // 模板名称
  charge_mode: number; // 配送计费方式
  sort: number; // 排序
  status: number; // 状态

  charges: MallTradeDeliveryExpressTemplateChargeRequest[]; // 运费列表
  frees: MallTradeDeliveryExpressTemplateFreeRequest[]; // 包邮列表
}

export interface MallTradeDeliveryExpressTemplateResponse {
  id: number; // 编号
  name: string; // 模板名称
  charge_mode: number; // 配送计费方式
  sort: number; // 排序
  status: number; // 状态
  creator: number; // 创建者ID
  create_time: string; // 创建时间
  updater: number; // 更新者ID
  update_time: string; // 更新时间

  charges: MallTradeDeliveryExpressTemplateChargeResponse[]; // 运费列表
  frees: MallTradeDeliveryExpressTemplateFreeResponse[]; // 包邮列表
}

export interface MallTradeDeliveryExpressTemplateQueryCondition
  extends PaginatedRequest {}

export const createMallTradeDeliveryExpressTemplate = (
  mall_trade_delivery_express_template: MallTradeDeliveryExpressTemplateRequest
): Promise<number> => {
  return api.post<number>(apis.create, mall_trade_delivery_express_template);
};

export const updateMallTradeDeliveryExpressTemplate = (
  mall_trade_delivery_express_template: MallTradeDeliveryExpressTemplateRequest
): Promise<void> => {
  return api.post<void>(apis.update, mall_trade_delivery_express_template);
};

export const deleteMallTradeDeliveryExpressTemplate = (
  id: number
): Promise<void> => {
  return api.post<void>(`${apis.delete}/${id}`);
};

export const getMallTradeDeliveryExpressTemplate = (
  id: number
): Promise<MallTradeDeliveryExpressTemplateResponse> => {
  return api.get<MallTradeDeliveryExpressTemplateResponse>(`${apis.get}/${id}`);
};

export const getBaseMallTradeDeliveryExpressTemplate = (
  id: number
): Promise<MallTradeDeliveryExpressTemplateResponse> => {
  return api.get<MallTradeDeliveryExpressTemplateResponse>(`${apis.get_base}/${id}`);
};

export const listMallTradeDeliveryExpressTemplate = (): Promise<
  Array<MallTradeDeliveryExpressTemplateResponse>
> => {
  return api.get<Array<MallTradeDeliveryExpressTemplateResponse>>(apis.list);
};

export const pageMallTradeDeliveryExpressTemplate = (
  condition: MallTradeDeliveryExpressTemplateQueryCondition
): Promise<PaginatedResponse<MallTradeDeliveryExpressTemplateResponse>> => {
  return api.get<PaginatedResponse<MallTradeDeliveryExpressTemplateResponse>>(
    apis.page,
    condition
  );
};

export const enableMallTradeDeliveryExpressTemplate = (
  id: number
): Promise<void> => {
  return api.post<void>(`${apis.enable}/${id}`);
};

export const disableMallTradeDeliveryExpressTemplate = (
  id: number
): Promise<void> => {
  return api.post<void>(`${apis.disable}/${id}`);
};

File: app/lib/services/mall_trade_delivery_express_template.dart
import 'package:synerunify/models/base.dart';

import '../utils/http_client.dart';
import '../utils/type_utils.dart';

const apis = {
  'create': '/mall/mall_trade_delivery_express_template/create', // 新增
  'update': '/mall/mall_trade_delivery_express_template/update', // 修改
  'delete': '/mall/mall_trade_delivery_express_template/delete', // 删除
  'get': '/mall/mall_trade_delivery_express_template/get', // 单条查询
  'list': '/mall/mall_trade_delivery_express_template/list', // 列表查询
  'page': '/mall/mall_trade_delivery_express_template/page', // 分页查询
  'enable': '/mall/mall_trade_delivery_express_template/enable', // 启用
  'disable': '/mall/mall_trade_delivery_express_template/disable', // 禁用
};

class MallTradeDeliveryExpressTemplateRequest {
  final int id; // 编号
  final String name; // 模板名称
  final int chargeMode; // 配送计费方式
  final int sort; // 排序
  final int status; // 状态
  

  MallTradeDeliveryExpressTemplateRequest({
    required this.id,
    required this.name,
    required this.chargeMode,
    required this.sort,
    required this.status,
    });

  factory MallTradeDeliveryExpressTemplateRequest.fromJson(Map<String, dynamic> json) {
    return MallTradeDeliveryExpressTemplateRequest(
      id: json['id'] as int,
      name: json['name'] as String,
      chargeMode: json['charge_mode'] as int,
      sort: json['sort'] as int,
      status: json['status'] as int,
      );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'charge_mode': chargeMode,
      'sort': sort,
      'status': status,
      };
  }
}

class MallTradeDeliveryExpressTemplateQueryCondition extends PaginatedRequest {
  MallTradeDeliveryExpressTemplateQueryCondition({
    required int page,
    required int size,
    String? keyword,
    String? sortField,
    String? sort,
    String? filterField,
    String? filterOperator,
    String? filterValue,
  }) : super(
          page: page,
          size: size,
          keyword: keyword,
          sortField: sortField,
          sort: sort,
          filterField: filterField,
          filterOperator: filterOperator,
          filterValue: filterValue,
        );
}

class MallTradeDeliveryExpressTemplateResponse {
  final int id; // 编号
  final String name; // 模板名称
  final int chargeMode; // 配送计费方式
  final int sort; // 排序
  final int status; // 状态
  final int? creator; // 创建者ID
  final DateTime createTime; // 创建时间
  final int? updater; // 更新者ID
  final DateTime updateTime; // 更新时间
  MallTradeDeliveryExpressTemplateResponse({
    required this.id,
    required this.name,
    required this.chargeMode,
    required this.sort,
    required this.status,
    this.creator,
    required this.createTime,
    this.updater,
    required this.updateTime,
    });

  factory MallTradeDeliveryExpressTemplateResponse.fromJson(Map<String, dynamic> json) {
    return MallTradeDeliveryExpressTemplateResponse(
      id: json['id'] as int,
      name: json['name'] as String,
      chargeMode: json['charge_mode'] as int,
      sort: json['sort'] as int,
      status: json['status'] as int,
      creator: json['creator'] as int?,
      createTime: json['create_time'] as DateTime,
      updater: json['updater'] as int?,
      updateTime: json['update_time'] as DateTime,
      );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'charge_mode': chargeMode,
      'sort': sort,
      'status': status,
      'creator': creator,
      'create_time': createTime,
      'updater': updater,
      'update_time': updateTime,
      };
  }
}

class MallTradeDeliveryExpressTemplateService {

    final HttpClient _httpClient = HttpClient();

    Future<ApiResponse<int>> createMallTradeDeliveryExpressTemplate(MallTradeDeliveryExpressTemplateRequest mallTradeDeliveryExpressTemplate) async {
        return await _httpClient.post<int>(apis['create']!, data: mallTradeDeliveryExpressTemplate);
    }

    Future<ApiResponse<int>> updateMallTradeDeliveryExpressTemplate(MallTradeDeliveryExpressTemplateRequest mallTradeDeliveryExpressTemplate) async {
        return await _httpClient.post<int>(apis['update']!, data: mallTradeDeliveryExpressTemplate);
    }

    Future<ApiResponse<void>> deleteMallTradeDeliveryExpressTemplate(int id) async {
        return await _httpClient.post<void>('${apis['delete']!}/$id');
    }

    Future<ApiResponse<MallTradeDeliveryExpressTemplateResponse>> getMallTradeDeliveryExpressTemplate(int id) async {
        return await _httpClient.get<MallTradeDeliveryExpressTemplateResponse>('${apis['get']!}/$id');
    }

    Future<ApiResponse<List<MallTradeDeliveryExpressTemplateResponse>>> listMallTradeDeliveryExpressTemplate() async {
        return await _httpClient.get<List<MallTradeDeliveryExpressTemplateResponse>>(apis['list']!);
    }

    Future<ApiResponse<PaginatedResponse<MallTradeDeliveryExpressTemplateResponse>>> pageMallTradeDeliveryExpressTemplate(MallTradeDeliveryExpressTemplateQueryCondition condition) async {
        return await _httpClient.get<PaginatedResponse<MallTradeDeliveryExpressTemplateResponse>>(apis['page']!, queryParameters: condition.toJson());
    }
    
    Future<ApiResponse<void>> enableMallTradeDeliveryExpressTemplate(int id) async {
      return await _httpClient.post<void>('${apis['enable']!}/$id');
    }

    Future<ApiResponse<void>> disableMallTradeDeliveryExpressTemplate(int id) async {
      return await _httpClient.post<void>('${apis['disable']!}/$id');
    }
}